tools/cpp_build/libtorch/CMakeLists.txt

```diff
@@ -141,51 +139,34 @@ TARGET_LINK_LIBRARIES(torch
   ${ATEN_LIBRARY}
 )
 
-TARGET_INCLUDE_DIRECTORIES(torch
-  PUBLIC
+set(COMMON_INCLUDES
   "${ATEN_INCLUDE_DIR}"
   "${ATEN_INCLUDE_DIR}/TH"
+  "${ATEN_BUILD_INCLUDE_DIR}"
+  "${ATEN_BUILD_PATH}/src/TH"
   "${CMAKE_CURRENT_SOURCE_DIR}"
+  "${CUDA_INCLUDE_DIRS}")
+
+TARGET_INCLUDE_DIRECTORIES(torch
+  PUBLIC
+  "${COMMON_INCLUDES}"
   "${TORCH_SRC_DIR}/../"
-  "${TORCH_SRC_DIR}/../third_party/nanopb"
-  "${CUDA_INCLUDE_DIRS}"
-  )
+  "${TORCH_SRC_DIR}/csrc/api/include"
+  "${TORCH_SRC_DIR}/../third_party/nanopb")
 
 SET_TARGET_PROPERTIES(torch PROPERTIES VERSION 1 SOVERSION 1)
 
 if(NOT ${CMAKE_VERSION} VERSION_LESS "3.1")
   SET_PROPERTY(TARGET torch PROPERTY CXX_STANDARD 11)
 endif(NOT ${CMAKE_VERSION} VERSION_LESS "3.1")
 
-install(DIRECTORY "${TORCH_SRC_DIR}/csrc"
-        DESTINATION ${TORCH_INSTALL_INCLUDE_DIR}/torch
-        FILES_MATCHING PATTERN "*.h")
-
-INSTALL(TARGETS torch
```

Comment:
While I agree with building against the ATen-build tree rather than relying on the Install tree, I would keep the install rules for the `libtorch` target. `ATen` has them too, they're only triggered by `make install`, they have no impact otherwise, but they can be useful for shipping / packaging.

Reply:
Sounds good